### ISSUE TITLE
IP address discovery can announce IPv6 localhost [::1] (on DEVELOPER builds)

### ISSUE THREAD

**Issue by m-schmoock** [ElementsProject/lightning#5234]
### Issue and Steps to Reproduce
Some of my peers report an local `remote_addr` of mine being for example:
```
   lightningd: Peer says it sees our address as: [::1]:44386
```
The current code will then lead to a `node_annoucement` announcing `[::1]:9375`.
Seem to happen on IPv6 only. This is not critical, but ugly for the time being.

Update: Can only on `DEVELOPER` builds. Needs to be fixed anyway

**Comment by m-schmoock** [comment]
If one configures with `--enable-developer` the node can send out local addresses for `remote_addr` (for testing). This is only one part of the explanation, because a node should clearly not fill in `::1` for a connection from a public address.

**Comment by m-schmoock** [comment]
@rustyrussell Its not as bad, turns out this can only happen on developer builds, which mine is.
I just need to change that to to --dev-allow-localhost and patch the tests

`connectd/peer_exchange_initmsg.c   peer_init_received  103ff`

```C
#if DEVELOPER		/* ignore private addresses (non-DEVELOPER builds) */
			if (address_routable(tlvs->remote_addr, true))
#else
			if (address_routable(tlvs->remote_addr, false))
#endif /* DEVELOPER */
```

**Comment by m-schmoock** [comment]
Resolved by #5244
